pkg/threescale/types.go

```diff
@@ -0,0 +1,33 @@
+package threescale
+
+import (
+	"net"
+	"net/http"
+
+	prometheus "github.com/3scale/3scale-istio-adapter/pkg/threescale/metrics"
+	"google.golang.org/grpc"
+)
+
+// Server interface - specifies the interface for gRPC server/adapter
+type Server interface {
+	Addr() string
+	Close() error
+	Run(shutdown chan error)
+}
+
+// Threescale contains the Listener and the server
+type Threescale struct {
+	listener net.Listener
+	server   *grpc.Server
+	client   *http.Client
+	conf     *AdapterConfig
+}
```

Comment:
Not sure if it's more idiomatic to group these together in golang, I personally prefer to have the definition next to the implementation of methods, perhaps one struct + methods per file (as in Java) to have all the information together for easy reference, but I don't have a strong opinion in this particular case (just FYI in case you didn't consider alternatives).

Reply:
Well this is not uncommon in go. Even the stdlib uses type files. The reason I have done this is looking ahead slightly. 

I spoke to @jmprusi who has a use cause for decoupling the actual so I expect some refactoring, but I'm ok with putting these back where they were if you like. Once the type methods aren't spread across files I'm fine either way.

Reply:
No, it's ok this way, I just FYI'ed it. :)

Decoupling this from it's context... well, it's going to cost a bit. Let @jmprusi volunteer to extract the logic out and make it available for us as a shared library. :D

Reply:
@jmprusi has just been volunteered. Thanks @jmprusi  👍 🥇